src/main/java/uk/ac/manchester/tornado/drivers/spirv/levelzero/samples/TestLevelZeroPowerMonitor.java

```diff
@@ -0,0 +1,129 @@
+        result = driver.zeDriverGetProperties(driverHandler, 0, driverProperties);
+        LevelZeroUtils.errorLog("zeDriverGetProperties", result);
+
+        System.out.println("Driver Version: " + driverProperties.getDriverVersion());
+
+        ZeAPIVersion apiVersion = new ZeAPIVersion();
+        result = driver.zeDriverGetApiVersion(driverHandler, 0, apiVersion);
+        LevelZeroUtils.errorLog("zeDriverGetApiVersion", result);
+
+        System.out.println("Level Zero API Version: " + apiVersion);
+
+        // ============================================
+        // Query device properties
+        // ============================================
+        LevelZeroDevice device = driver.getDevice(driverHandler, 0);
+
+        ZeDeviceProperties deviceProperties = new ZeDeviceProperties();
+        result = device.zeDeviceGetProperties(device.getDeviceHandlerPtr(), deviceProperties);
+        LevelZeroUtils.errorLog("zeDeviceGetProperties", result);
+        System.out.println("Device   : " + deviceProperties.getName());
+        System.out.println("Type     : " + deviceProperties.getType());
+        System.out.println("Vendor ID: " + deviceProperties.getVendorId());
+
+        // get all device handles
+        long[] devicePointers = deviceHandler.getDevicePointers();
+        
+        LevelZeroPowerMonitor powerUsage = new LevelZeroPowerMonitor();
+        
+        // cast all device handles to sysman handles    
+        long[] sysmanDevices = powerUsage.ZesInit(devicePointers);
+        
+        // Check if power query is possible for any devices - any iGPU devices?
+        int checkResult = powerUsage.checkPowerQueryPossible(sysmanDevices);
+
+        if (checkResult == -1) {
+            System.out.println("Power query is not possible. No sysman devices found.");
+        } else if (checkResult == -2 ) {
+            System.err.println("Power query is not possible. All devices are Integrated GPUs.");
+        }
+        else {
+            // Success - There are query-able devices (at least one dGPU)
+        }
+
+        // Get only the devices that we can query (dGPUs)
+        long[] devicesToQuery = powerUsage.getSysmanDevicesToQuery(sysmanDevices);
+
+        // Testing with the 1st Sysman device that we can query
+        List<ZesPowerEnergyCounter> initialEnergyCounters = powerUsage.getEnergyCounters(devicesToQuery[0]);
+
+        // wait 5 seconds - this is what we're measuring
+        try {
+            Thread.sleep(5000);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+
+        List<ZesPowerEnergyCounter> finalEnergyCounters = powerUsage.getEnergyCounters(devicesToQuery[0]);
+        
+        double powerUsage_mW = powerUsage.calculatePowerUsage(initialEnergyCounters, finalEnergyCounters);
+
+        System.out.println("Power Usage: " + powerUsage_mW + " mW");
```

Comment:
I tested it with iGPU and I see this message:
```bash
Level-ZERO JNI Library - TestSysmanPowerFunctions
ZE_LOADER_DEBUG_TRACE:Using Loader Library Path: 
ZE_LOADER_DEBUG_TRACE:Tracing Layer Library Path: libze_tracing_layer.so.1
Driver Version: 17003559
Level Zero API Version: 1.3
Device   : Intel(R) Iris(R) Xe Graphics
Type     : ZE_DEVICE_TYPE_GPU
Vendor ID: 32902
Error at zesDeviceEnumPowerDomains(hSysmanDevice, &numPowerDomains, nullptr): IsIntegratedGPU: 49
Exit with Error Code: 0x78000001
Power query is not possible. All devices are Integrated GPUs.
Error at zesDeviceEnumPowerDomains(hSysmanDevice, &numPowerDomains, nullptr): IsIntegratedGPU: 49
Exit with Error Code: 0x78000001
Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: Index 0 out of bounds for length 0
	at uk.ac.manchester.tornado.drivers.spirv.levelzero.samples.TestLevelZeroPowerMonitor.main(TestLevelZeroPowerMonitor.java:114)
```

I suggest to add the following condition in the test, as suggested.

```suggestion
        // Testing with the 1st Sysman device that we can query
        if (devicesToQuery.length > 0) {
            List<ZesPowerEnergyCounter> initialEnergyCounters = powerUsage.getEnergyCounters(devicesToQuery[0]);

            // wait 5 seconds - this is what we're measuring
            try {
                Thread.sleep(5000);
            } catch (InterruptedException e) {
                e.printStackTrace();
            }

            List<ZesPowerEnergyCounter> finalEnergyCounters = powerUsage.getEnergyCounters(devicesToQuery[0]);

            double powerUsage_mW = powerUsage.calculatePowerUsage(initialEnergyCounters, finalEnergyCounters);


            System.out.println("Power Usage: " + powerUsage_mW + " mW");
        }
```